### PR TITLE
use monotonic timer

### PR DESCRIPTION
**Pull request by elwerene**

### CODE REVIEW

File: demos/nrf52-beacon/src/main.rs [hunk 1]
```diff
@@ -81,19 +61,25 @@ const APP: () = {
...
-    fn timer1(ctx: timer1::Context) {
-        // Acknowledge event so that the interrupt doesn't keep firing
-        ctx.resources.beacon_timer.events_compare[0].reset();
-
+    #[task(schedule = [update], resources = [beacon, radio])]
+    fn update(ctx: update::Context) {
         ctx.resources.beacon.broadcast(ctx.resources.radio);
+
+        ctx.schedule
+            .update(ctx.scheduled + 1_066_666.cycles())
```
Comment (jonas-schievink) [inline demos/nrf52-beacon/src/main.rs:75]: Can you add a comment explaining where the 1066666 cycles number comes from?

Comment (elwerene) [reply demos/nrf52-beacon/src/main.rs:75]: sure

Comment (elwerene) [reply demos/nrf52-beacon/src/main.rs]: good catch, it was wrong by factor 10 anyways :facepalm:

File: demos/nrf52-beacon/src/main.rs [hunk 2]
```diff
@@ -81,19 +61,25 @@ const APP: () = {
...
-        // Acknowledge event so that the interrupt doesn't keep firing
-        ctx.resources.beacon_timer.events_compare[0].reset();
-
+    #[task(schedule = [update], resources = [beacon, radio])]
+    fn update(ctx: update::Context) {
         ctx.resources.beacon.broadcast(ctx.resources.radio);
+
+        ctx.schedule
+            .update(ctx.scheduled + 10_666_666.cycles())
+            .ok(); // about 3 times per second as the nrf52 runs with 32 MHz
```
Comment (jonas-schievink) [inline demos/nrf52-beacon/src/main.rs:76]: Hmm, the core speed is 64 MHz though?

Comment (elwerene) [reply demos/nrf52-beacon/src/main.rs]: The clock is at 32 Mhz. I updated the comment.

Comment (jonas-schievink) [reply demos/nrf52-beacon/src/main.rs]: CYCCNT counts at the core frequency from what I understand, which is 64 MHz, right?

Comment (elwerene) [reply demos/nrf52-beacon/src/main.rs]: Yes yes, you are right :facepalm: I had a blinky test with a delay of 32_000_000 and thought it must be 32 MHz as it was blinking once a second. Well that was very wrong :)

Comment (elwerene) [reply demos/nrf52-beacon/src/main.rs]: Fixed it, sorry

**Comment by dbrgn** [comment]
This looks nice, but keep in mind that CYCCNT only works when the debug peripheral is enabled. Otherwise, a custom monotonic impl is required (e.g. https://github.com/rtic-rs/rtic-examples/blob/master/rtic_v5/monotonic_nrf52/src/monotonic_nrf52.rs).

**Pull request merged**